indico/cli/i18n.py

```diff
@@ -113,62 +244,241 @@ def _wrapper(**kwargs):
+
+    @_common_translation_options(**kwargs)
+    def wrapper(**kwargs):
+        func(babel_cmd=babel_cmd, **kwargs)
+
+    return cmd_group(wrapper)
+
+
+_make_command(compile_catalog, 'indico', 'compile_catalog', _command, require_js=False)
+_make_command(extract_messages, 'indico', 'extract_messages', _command)
+_make_command(update_catalog, 'indico', 'update_catalog', _command)
+_make_command(init_catalog, 'indico', 'init_catalog', _command, require_locale=True)
+
+_make_command(compile_catalog, 'plugins', 'compile_catalog', _command, require_js=False,
+              plugins=True)
+_make_command(extract_messages, 'plugins', 'extract_messages', _command, plugins=True)
+_make_command(update_catalog, 'plugins', 'update_catalog', _command, plugins=True)
+_make_command(init_catalog, 'plugins', 'init_catalog', _command, require_locale=True,
+              plugins=True)
+
+_make_command(compile_catalog, 'all-plugins', 'compile_catalog', _command, require_js=False,
+              all_plugins=True)
+_make_command(extract_messages, 'all-plugins', 'extract_messages', _command, all_plugins=True)
+_make_command(update_catalog, 'all-plugins', 'update_catalog', _command, all_plugins=True)
+_make_command(init_catalog, 'all-plugins', 'init_catalog', _command, require_locale=True,
+              all_plugins=True)
+
+
+@cli.command(short_help='Push .pot files to Transifex.')
+def push():
+    """Push .pot files to Transifex."""
+    try:
+        subprocess.run(['tx', 'push', '-s'], check=True)
+    except subprocess.CalledProcessError:
+        click.secho('Error pushing to transifex ', fg='red', bold=True, err=True)
+
+
+@cli.command(short_help='Pull translated .po files from Transifex.')
+@click.argument('languages', nargs=-1, required=False)
+def pull(languages):
+    """Pull translations from Transifex.
+
+    Pulls only the language codes specified in `languages` if present.
+    If no `languages`, it pulls all the languages currently available in the translations folder.
+    To pull a new language available on Transifex, simply pass the new language code in the argument.
     """
-    root_path = 'indico/translations'
+    if not languages:
+        languages = [d for d in os.listdir(TRANSLATIONS_DIR) if os.path.isdir(os.path.join(TRANSLATIONS_DIR, d))]
+    try:
+        for language in languages:
+            subprocess.run(['tx', 'pull', '-l', language, '-f'], check=True)
+            click.secho(f'Translations updated for {language}.', fg='green', bold=True)
+    except subprocess.CalledProcessError:
+        click.secho('Error pulling from transifex', fg='red', bold=True, err=True)
+    finally:
+        language_renames = {'zh_CN.GB2312': 'zh_Hans_CN'}  # other lang codes requiring rename should be added here
+        languages = set(languages) & set(language_renames.keys())
+        for code in languages:
+            if os.path.exists(os.path.join(TRANSLATIONS_DIR, code)):
```

Comment:
Probably not necessary to check if the path exists - if the pull was successful, the folder should exist

Reply:
I just realised that the zh_CN.GB2312 translations would not be pulled when just using `indico i18n pull` because it would try to pull by the local name in the directory so zh_Hans_CN. I'm going to fix this. 

But when pulling individual language codes that require rename, would you prefer to pull by the name available on indico or by the name on transifex? So `indico i18n pull zh_Hans_CN` or `indico i18n pull zh_CN.GB2312` ? @ThiefMaster, @tomasr8

Reply:
I decided to stick with the name in indico, so pulling with `indico i18n pull zh_Hans_CN`